packages/core/src/shared/utilities/messageUtil.ts

```diff
@@ -0,0 +1,31 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { CodeWhispererStreamingServiceException } from '@amzn/codewhisperer-streaming'
+import { getHttpStatusCode, getRequestId } from '../errors'
+
+export interface MessageErrorInfo {
+    errorMessage: string
+    statusCode?: number
+    requestId?: string
+}
+
+export function extractErrorInfo(error: any): MessageErrorInfo {
```

Comment:
Thanks for thinking about shared utils, but this is specific to CW client, so it should live in `packages/core/src/shared/clients/codewhispererChatClient.ts`.